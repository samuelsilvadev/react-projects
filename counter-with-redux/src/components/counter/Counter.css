.counter {
    display: flex;
    flex-wrap: wrap;
    margin-top: 50vh;
    margin-left: auto;
    margin-right: auto;
    max-width: 200px;
    text-align: center;
    transform: translateY(-50%);
    width: 100%;
}

.counter__number {
    font-size: 2rem;
    font-weight: bold;
    width: 100%;
}

.counter__button {
    flex-grow: 1;
    background-color: #1e88e5;
}

.counter__button:hover, .counter__button:focus {
    background-color: #1976d2;
}