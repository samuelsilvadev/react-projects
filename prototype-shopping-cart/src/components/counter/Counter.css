.counter {
    margin: 1rem 0;
    display: flex;
    justify-content: center;
}

.counter__quantity {
    min-width: 40px;
}

.counter__btn-increment {
	margin-right: 0.5rem;
}
