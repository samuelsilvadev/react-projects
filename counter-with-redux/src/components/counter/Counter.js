import React from 'react';
import { connect } from 'react-redux';

import './Counter.css';

const INCREMENT_TYPE = { type: 'INCREMENT' };
const DECREMENT_TYPE = { type: 'DECREMENT' };

class Counter extends React.PureComponent {
    render() { 
        const { counter } = this.props;

        return (
            <div className="counter card">
                <p className="counter__number">{ counter }</p>
                <button
                    className="counter__button btn waves-effect waves-light"
                    onClick={ this._handleDecrement }>
                    -
                </button>
                <button
                    className="counter__button btn waves-effect waves-light"
                    onClick={ this._handleIncrement }>
                    +
                </button>
            </div>
        );
    }

    _handleIncrement = () => {
        const { dispatch } = this.props; 

        dispatch(INCREMENT_TYPE);
    }

    _handleDecrement = () => {
        const { dispatch } = this.props; 

        dispatch(DECREMENT_TYPE);
    }
}

const mapStateToProps = (state) => {
    return {
        counter: state,
    };
}

export default connect(mapStateToProps)(Counter);