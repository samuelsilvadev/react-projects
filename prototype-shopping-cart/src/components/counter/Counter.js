import React from 'react';
import PropTypes from 'prop-types';

import './Counter.css';

class Counter extends React.Component {
    state = {
        value: this.props.value,
    }

    render() {
        return (
            <div className="counter">
                <span
                    className={this._getBadgeClasses()}>
                    {this._formatCount()}
                </span>
                <button
                    className="btn btn-primary btn-sm"
                    onClick={this._handleClickIncrementBtn}>
                    Increment
                </button>
            </div>
        );
    }

    _handleClickIncrementBtn = () => {
        const { value } = this.state;
        this.setState({
            value: value + 1,
        });
    }

    _formatCount() {
        const { value } = this.state;
        return (value === 0 ? 'Zero' : value);
    }

    _getBadgeClasses() {
        const { value } = this.state;
        return `counter__quantity m-2 badge badge-${value === 0 ? 'warning' : 'success'}`;
    }
}

Counter.propTypes = {
    value: PropTypes.number,
};

Counter.defaultProps = {
    value: 0,
};

export default Counter;