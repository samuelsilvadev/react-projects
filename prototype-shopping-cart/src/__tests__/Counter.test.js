import React from 'react';
import { shallow } from 'enzyme';

import Counter from './../components/counter/Counter';

describe('<Counter />', () => {
    it('should render correctly and match snapshot', () => {
        const CounterComponent = shallow(<Counter id={1} value={0} />);
        const instance = CounterComponent.instance();
        
        expect(instance.props.value).toBe(0);
        expect(instance.state.value).toBe(0);
        expect(CounterComponent).toMatchSnapshot();
    });

    it('should update state after `_handleClickIncrementBtn` be called', () => {
        const CounterComponent = shallow(<Counter id={1} value={1} />);
        const instance = CounterComponent.instance();
        
        expect(instance.props.value).toBe(1);
        expect(instance.state.value).toBe(1);

        instance._handleClickIncrementBtn();

        expect(instance.state.value).toBe(2);
    });

    it('should return correct value after `_formatCount` be called', () => {
        const CounterComponent = shallow(<Counter id={1} />);
        const instance = CounterComponent.instance();
        
        expect(instance._formatCount()).toBe('Zero');

        instance._handleClickIncrementBtn();

        expect(instance._formatCount()).toBe(1);
    });
});
