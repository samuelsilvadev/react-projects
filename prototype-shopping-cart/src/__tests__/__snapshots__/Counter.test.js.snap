// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should render correctly and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    value={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="counter__quantity m-2 badge badge-warning"
        >
          Zero
        </span>,
        <button
          className="btn btn-primary btn-sm"
          onClick={[Function]}
        >
          Increment
        </button>,
      ],
      "className": "counter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Zero",
          "className": "counter__quantity m-2 badge badge-warning",
        },
        "ref": null,
        "rendered": "Zero",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Increment",
          "className": "btn btn-primary btn-sm",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Increment",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="counter__quantity m-2 badge badge-warning"
          >
            Zero
          </span>,
          <button
            className="btn btn-primary btn-sm"
            onClick={[Function]}
          >
            Increment
          </button>,
        ],
        "className": "counter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Zero",
            "className": "counter__quantity m-2 badge badge-warning",
          },
          "ref": null,
          "rendered": "Zero",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Increment",
            "className": "btn btn-primary btn-sm",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Increment",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
